public/renderer.js

```diff
@@ -523,113 +534,157 @@ function getSectorBoundsAllBus (roomInSector) {
   if (y < 0) y -= 9
   let sx = x - (x % 10)
   let sy = y - (y % 10)
-  let start = { x: sx , y: sy }
-  let end = { x: sx + 10, y: sy + 10 }
+  let start = {x: sx, y: sy}
+  let end = {x: sx + 10, y: sy + 10}
   if (x < 0 && y > -1) {
-      //wxsx
-      start.x -= 1
-      end.y += 1
+    //wxsx
+    start.x -= 1
+    end.y += 1
   } else if (x < 0 && y < 0) {
-      //wxnx
-      start.x -= 1
-      start.y -= 1
+    //wxnx
+    start.x -= 1
+    start.y -= 1
   } else if (x > -1 && y < 0) {
-      //exnx
-      start.y -= 1
-      end.x += 1
+    //exnx
+    start.y -= 1
+    end.x += 1
   } else {
-      //exsx
-      end.x += 1
-      end.y += 1
+    //exsx
+    end.x += 1
+    end.y += 1
   }
-  return { start, end }
+  return {start, end}
 }
 
 function makeRespawnSectorWall (room, borderSide, decayTime) {
-  let x, y;
+  let x, y, x2, y2
+  // Edges
   for (let i = 0; i < 50; i++) {
-      switch(borderSide) {
-          //borderSide is related to the sector side not the room side
-          case 'left':
-              x = 49;
-              y = i;
-              break;
-          case 'right':
-              x = 0;
-              y = i;
-              break;
-          case 'top':
-              x = i;
-              y = 49;
-              break;
-          case 'bottom':
-              x = i;
-              y = 0;
-              break;
-      }
-      if (x !== undefined && y !== undefined) {
-          room.objects.push({ type: 'constructedWall', room: room.room, x, y, decayTime: { timestamp: decayTime } });
-      }
+    switch (borderSide) {
+      //borderSide is related to the sector side not the room side
+      case 'left':
+        x = 30
+        y = i
+        break
+      case 'right':
+        x = 20
+        y = i
+        break
+      case 'top':
+        x = i
+        y = 30
+        break
+      case 'bottom':
+        x = i
+        y = 20
+        break
+    }
+    if (x !== undefined && y !== undefined) {
+      room.objects.push({type: 'constructedWall', room: room.room, x, y, decayTime: {timestamp: decayTime}})
+    }
+  }
+  // Corners
+  for (let i = 0; i < 20; i++) {
+    switch (borderSide) {
+      case 'bottomLeft':
+        x = 30
+        y = i
+        x2 = 30 + i
+        y2 = 20
+        break
+      case 'topLeft':
+        x = 30 + i
+        y = 30
+        x2 = 30
+        y2 = 30 + i
+        break
+      case 'bottomRight':
+        x = i
+        y = 20
+        x2 = 20
+        y2 = i
+        break
+      case 'topRight':
+        x = i
+        y = 30
+        x2 = 20
+        y2 = 30 + i
+        break
+    }
+    if (x !== undefined && y !== undefined && x < 50 && y < 50) {
+      room.objects.push({type: 'constructedWall', room: room.room, x, y, decayTime: {timestamp: decayTime}})
+    }
+    if (x2 !== undefined && y2 !== undefined && x2 < 50 && y2 < 50) {
+      room.objects.push({type: 'constructedWall', room: room.room, x2, y2, decayTime: {timestamp: decayTime}})
```

Comment:
assignment error here, should be:
room.objects.push({type: 'constructedWall', room: room.room, x: x2, y: y2, decayTime: {timestamp: decayTime}})